graylog2-server/src/main/java/org/graylog2/inputs/codecs/JsonPathCodec.java

```diff
@@ -105,6 +134,50 @@ protected String buildShortMessage(Map<String, Object> fields) {
         return shortMessage.toString();
     }
 
+    public Map<String, Object> useFlattener(String json) throws JsonTypeException {
+        Map<String, Object> map = new HashMap<>();
+        try {
+            flatten("", OBJECT_MAPPER.readTree(json), map);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
You should not just print the stack trace here, but instead use the logger (`LOG`) and/or re-throw the exception.